cs-abbyy/src/main/java/io/cloudslang/content/abbyy/services/ProcessImageService.java

```diff
@@ -156,7 +157,11 @@ protected void handleTaskCompleted(ProcessImageInput request, AbbyyResponse resp
                         String pdfUrl = response.getResultUrls().get(indexOfPdfUrl);
                         results.put(OutputNames.PDF_URL, pdfUrl);
                         if (request.getDestinationFile() != null) {
-                            downloadFromUrl(request, pdfUrl, exportFormat);
+                            String targetPath = getTargetPath(request, exportFormat);
+                            if (new File(targetPath).exists()) {
+                                throw new FileAlreadyExistsException(targetPath);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Exception [FileAlreadyExistsException] is explicitly thrown on this line but is caught in the outer catch clause with generic exception on line [168].  Consider adding a catch clause of this specific exception in the outer try statement.

Reply:
The idea there was to catch any exception on the current export format, save it in the result in order to inform the user about the failure, and continue trying to download the next export formats